sklearn/metrics/ranking.py

```diff
@@ -95,8 +105,17 @@ def auc(x, y, reorder=False):
             if np.all(dx <= 0):
                 direction = -1
             else:
-                raise ValueError("Reordering is not turned on, and "
-                                 "the x array is not increasing: %s" % x)
+                maxpos = dx.argmax()
+                minpos = dx.argmin()
+                raise ValueError("x is neither increasing nor decreasing "
```

Comment:
I think this message is a bit obscure to a reader of the code, more so than to someone who receives the error by surprise. It deserves a comment, perhaps a reference to the issue.